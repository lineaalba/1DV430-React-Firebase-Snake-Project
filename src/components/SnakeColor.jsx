/**
 * @author Filippa Jakobsson
 * @version 1.0
 */

import React from 'react'
import Button from 'react-bootstrap/Button'

/**
* Function to change snake color.
*/
const SnakeColor = () => {
  const snakeDots = document.querySelectorAll('.snake__dot')
  const snakeColors = document.querySelectorAll('.snake__color')
    
 
  for (let i = 0; i < snakeColors.length; i++) {
    let color

    snakeColors[i].addEventListener('click', e => {
      switch (snakeColors[i].textContent) {
        case 'Yellow':
          color = 'yellow'
          break
        case 'Red':
          color = 'red'
          break
        case 'Green':
          color = 'rgb(11, 184, 34)'
          break
        case 'Blue':
          color = 'blue'
          break
        case 'Pink':
          color = 'rgb(252, 97, 123)'
          break
        default:
          color = 'rgb(11, 184, 34)' 
      }    
        for (let i = 0; i < snakeDots.length; i++) {
          snakeDots[i].style.background = color
        }
    })
  }
  
    return (
        <div className="dropdown">
          <Button data-testid="snakecolor" className="change__color" variant="outline-dark">Change snake color</Button>
          <div onClick={SnakeColor} className="dropdown__content">
            <p className="snake__color">Yellow</p>
            <p className="snake__color">Red</p>
            <p className="snake__color">Green</p>
            <p className="snake__color">Blue</p>
            <p className="snake__color">Pink</p>
          </div>
        </div>
    )
}

export default SnakeColor