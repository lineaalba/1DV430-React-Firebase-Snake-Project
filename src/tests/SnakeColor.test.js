import React from 'react'
import { shallow } from 'enzyme'
import SnakeColor from '../components/SnakeColor'

describe('Tests the SnakeColor component', () => {

  it('matches snapshot', () => {
    const wrapper = shallow(<SnakeColor />)
    expect(wrapper.html()).toMatchSnapshot()
  })
})