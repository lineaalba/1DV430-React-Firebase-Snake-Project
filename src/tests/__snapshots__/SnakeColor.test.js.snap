// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the SnakeColor component matches snapshot 1`] = `"<div class=\\"dropdown\\"><button data-testid=\\"snakecolor\\" type=\\"button\\" class=\\"change__color btn btn-outline-dark\\">Change snake color</button><div class=\\"dropdown__content\\"><p data-testid=\\"yellow\\" class=\\"snake__color\\">Yellow</p><p data-testid=\\"red\\" class=\\"snake__color\\">Red</p><p data-testid=\\"green\\" class=\\"snake__color\\">Green</p><p data-testid=\\"blue\\" class=\\"snake__color\\">Blue</p><p data-testid=\\"pink\\" class=\\"snake__color\\">Pink</p></div></div>"`;
